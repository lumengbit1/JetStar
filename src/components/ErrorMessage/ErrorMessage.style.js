import styled from 'styled-components';

export const Command = styled.div`
  color: red;
`;
