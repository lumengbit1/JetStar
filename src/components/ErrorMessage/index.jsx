import React from 'react';
import { useSelector } from 'react-redux';
import { Command } from './ErrorMessage.style';

const ErrorMessage = () => {
  const errorMessage = useSelector((state) => state.reducer.errorMessage);

  if (errorMessage.length === 0) return null;

  return (
    <Command aria-label="error-list">
      {errorMessage}
    </Command>
  );
};

export default ErrorMessage;
