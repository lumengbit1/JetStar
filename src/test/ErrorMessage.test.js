import React from 'react';
import 'jest-styled-components';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';

import { render } from '@testing-library/react';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store'
import ErrorMessage from '../components/ErrorMessage';

const mockStore = configureMockStore([thunk]);
const store = mockStore();

describe('ErrorMessage Test Cases', () => {
  it('1: Component shoule be rendered correctly', async () => {
    const store = mockStore({
      reducer: { errorMessage: 'error message' }
    });

    const { container } = render(
      <Provider store={store}>
        <ErrorMessage />
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});