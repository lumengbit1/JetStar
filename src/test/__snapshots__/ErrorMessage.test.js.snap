// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage Test Cases 1: Component shoule be rendered correctly 1`] = `
.c0 {
  color: red;
}

<div>
  <div
    aria-label="error-list"
    class="c0"
  >
    error message
  </div>
</div>
`;
